Public/Src/IDE/Generator/CsprojFile.cs

```diff
@@ -364,10 +369,28 @@ private void IterateCscArguments(Project project, PipData arguments, bool isNest
                             };
                             break;
                         default:
-                            if (strValue.StartsWith("/target:", StringComparison.OrdinalIgnoreCase))
+                            const string Target = "/target:";
+                            const string Define = "/define:";
+                            const string Reference = "/r:";
+
+                            if (strValue.StartsWith(Target, StringComparison.OrdinalIgnoreCase))
+                            {
+                                project.SetProperty("OutputType", strValue.Substring(Target.Length));
+                                break;
+                            }
+
+                            if (strValue.StartsWith(Define, StringComparison.OrdinalIgnoreCase))
+                            {
+                                project.SetProperty("DefineConstants", strValue.Substring(Define.Length).Trim('"'));
+                                break;
+                            }

```

Comment:
why not include these as cases in the switch statement above?

Reply:
The switch cases work when args are represented with 2 pip fragments: `/define:` and then another string fragment.  Because of escaping of whatever other reason, when we pass this argument to csc.exe, it is represented as a single StringLiteral fragment (`/define:"Const1;Const2"`) which is why a case for `/define:` would never get matched.

Reply:
In fact, there is already 
```
                        case "/define:":
                            action = (obj) => project.SetProperty("DefineConstants", (string)obj);
                            break;
```
present in the code, but that doesn't do the trick